cloudsync/provider.py

```diff
@@ -26,15 +27,47 @@
 
 CONNECTION_NOT_NEEDED = "connection-not-needed"
 
-__all__ = ["Provider", "Namespace", "Creds", "Hash", "Cursor", "CONNECTION_NOT_NEEDED"]
+__all__ = ["Provider", "Namespace", "NamespaceBase", "Creds", "Hash", "Cursor", "CONNECTION_NOT_NEEDED"]
 
 
+# for backward compatibility
```

Comment:
this should derive from NamespaceBase, and just have a setter for is_parent.

Reply:
the problem here is that OneDriveProvider has classes that derive from Namespace(NamedTuple) and there is a very specific way of doing so that breaks when I take NamedTuple out of the class hierarchy...

```python
class Site(Namespace):
  def __new__(cls, name, site_id, drives=None, cached=False): 
  ...
```